### ISSUE TITLE
Method chaining with anonymous functions breaks the indentation sniffer

### ISSUE THREAD

**Issue by aminin** [squizlabs/PHP_CodeSniffer#1053]
With the code

```php
<?php

namespace Example\NSpace;

class ExampleClass
{
    public static function isIndentedCorrectly($bar)
    {
        $foo = $bar->filter(function ($a) {
            return $a->isActive();
        })->map(function ($a) {
            return $a->getTitle();
        })->toArray();

        if (!empty($foo)) {
            echo 'Success!';
        }
    }
}
```
in the `foo.php` file, the command

```
phpcs --standard=PSR2 --encoding=utf-8 -n -s foo.php
```

produces the following result

```
FILE: /home/anton/workspace/omlook/omlook/foo.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 15 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found
    |       |     8 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 17 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found
    |       |     8 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 24ms; Memory: 4.75Mb
```

Yet the code looks indented corectly.

**Comment by gsherwood** [comment]
This was originally reported as bug #961 and fixed in master. It will be in the next release. Thanks for reporting it anyway.
